api/v1beta1/oscmachine_webhook.go

```diff
@@ -64,15 +64,6 @@ func (m *OscMachine) ValidateUpdate(oldRaw runtime.Object) error {
 	var allErrs field.ErrorList
 	old := oldRaw.(*OscMachine)
 
-	oscMachineLog.Info("validate update old imageId", "old imageId", old.Spec.Node.Vm.ImageId)
-	oscMachineLog.Info("validate update imageId", "imageId", m.Spec.Node.Vm.ImageId)
-	if !reflect.DeepEqual(m.Spec.Node.Vm.ImageId, old.Spec.Node.Vm.ImageId) {
-		allErrs = append(allErrs,
-			field.Invalid(field.NewPath("spec", "imageId"),
-				m.Spec.Node.Vm.ImageId, "field is immutable"),
-		)
-	}
-
 	oscMachineLog.Info("validate update old vmType", "old vmType", old.Spec.Node.Vm.VmType)
```

Comment:
Explain why you remove this part of the code

Reply:
We can only update value which are in cluster-template.yaml.